Backpack-SwiftUI/FlightLeg/Classes/BPKFlightLeg.swift

```diff
@@ -0,0 +1,177 @@
+import SwiftUI
+import Backpack_Common
+
+struct BPKFlightLeg: View {
+    private let airlineLogoSize: CGFloat = 24.0
+
+    private let departureArrivalTime: String
+    private let nextDayArrival: String?
+    private let flightDescription: AttributedString
+    private let stopsInfo: String
+    private let highlightStopsInfo: Bool
+    private let duration: String
+    private let operatedBy: String?
+    private let warning: String?
+    @State private var carrierLogo: Image?
+
+    // String-based init
+    public init(
+        departureArrivalTime: String,
+        nextDayArrival: String? = "",
+        flightDescription: String,
+        stopsInfo: String,
+        highlightStopsInfo: Bool,
+        duration: String,
+        operatedBy: String? = "",
+        warning: String? = "",
+        carrierLogo: UIImage? = nil
+    ) {
+        self.departureArrivalTime = departureArrivalTime
+        self.nextDayArrival = nextDayArrival
+        self.flightDescription = AttributedString(flightDescription)
+        self.stopsInfo = stopsInfo
+        self.highlightStopsInfo = highlightStopsInfo
+        self.duration = duration
+        self.operatedBy = operatedBy
+        self.warning = warning
+        if let carrierLogo = carrierLogo {
+            self._carrierLogo = State(initialValue: Image(uiImage: carrierLogo))
+        } else {
+            self._carrierLogo = State(initialValue: nil)
+        }
+    }
+
+    // AttributedString-based init
+    public init(
+        departureArrivalTime: String,
+        nextDayArrival: String? = "",
+        flightDescription: AttributedString,
+        stopsInfo: String,
+        highlightStopsInfo: Bool,
+        duration: String,
+        operatedBy: String? = "",
+        warning: String? = "",
+        carrierLogo: UIImage? = nil
+    ) {
+        self.departureArrivalTime = departureArrivalTime
+        self.nextDayArrival = nextDayArrival
+        self.flightDescription = flightDescription
+        self.stopsInfo = stopsInfo
+        self.highlightStopsInfo = highlightStopsInfo
+        self.duration = duration
+        self.operatedBy = operatedBy
+        self.warning = warning
+        if let carrierLogo = carrierLogo {
+            self._carrierLogo = State(initialValue: Image(uiImage: carrierLogo))
+        } else {
+            self._carrierLogo = State(initialValue: nil)
+        }
+    }
+
+    var body: some View {
+        BPKDynamicStack(horizontalAlignment: .leading, verticalAlignment: .top, spacing: .base) {
+            createCarrierLogo()
+            createFlightInfo()
+                .frame(
+                    maxWidth: .infinity,
+                    alignment: .leading
+                )
+            createDurationInfo()
+        }
+    }
+
+    @ViewBuilder
+    private func createCarrierLogo() -> some View {
```

Comment:
could you change the name of these funcitons to remove the word 'create' from them? name them as if you were composing this view as any other view. just `carrierLogo` should be good! also if it does not receive params, it should probably be a `var` instead.
A slighly better way would be to make this a little struct internal to the BPKFlightLeg to contain this view.
```swift
struct CarrierLogo: View {
 ...
}